lib/api/repository/auth_repository.dart

```diff
@@ -0,0 +1,41 @@
+import '../../env.dart';
+import '../../model/auth_model.dart';
+import '../exception/network_exceptions.dart';
+import '../request/login_request.dart';
+import '../service/auth_service.dart';
+
+const _passwordType = 'password';
+
+abstract class AuthRepository {
+  Future<AuthModel> login({
```

Comment:
Can we rename it to `signIn` for consistency? Since we are using `sign in` for the authentication steps.